KFSPTS-29350 Upgrade to the 08/30/2023 version of financials

cah292 opened the pull request:
There are changes for this upgrade in cu-kfs, nonprod-sql and cu-kfs-upgrades. Please make sure all 3 PRs are good to go before merging.

NOTE: Our "secretsmanager" dependency is pulling in additional artifacts that would impact the FINP-9994 validation. That dependency needs to remain in place until we're ready to complete KFSPTS-33874, which removes the dependent code. If you think KFSPTS-33874 should be implemented in conjunction with (or just prior to) the 2023-08-30 upgrade, please let me know.

db483 commented:
@cah292 Please could you confirm if the code for the secrets manager implementation is causing an issue with this upgrade? more specifically FINP-9994 ? Thank you

db483 reviewed (changes requested):
Changes look good but I had some questions and things to consider. Thank you.

db483 commented on pom.xml:
```diff
@@ -32,23 +32,27 @@
-        <opencsv.version>5.7.1</opencsv.version>
+        <opencsv.version>5.8</opencsv.version>
         <plexus-utils.version>3.0.24</plexus-utils.version>
         <mockito.version>4.9.0</mockito.version>
```
Thank you for the reminder. I think we thought https://jira.cornell.edu/browse/KFSPTS-27119 would be taking care of this.

cah292 replied:
Thanks for reminding me about the specific JIRA number. I would suggest we use KFSPTS-27119 to handle the Mockito upgrade, then.

db483 replied:
That makes sense, thank you.

cah292 commented on src/main/resources/edu/cornell/kfs/module/cg/businessobject/datadictionary/Agency.xml:
```diff
@@ -16,23 +16,18 @@
-  
+
   <bean id="Agency-fullName" parent="Agency-fullName-parentBean">
-  	<property name="maxLength" value="150"/>
```
From what I can tell, KFSPTS-9458 increased this field's max length to 150 in both the application and the database. Now that base code is increasing it to 300 for this upgrade, I don't anticipate any problems, unless there are any external systems that rely upon this data but only expect a 150 character length.

cah292 commented on src/main/java/edu/cornell/kfs/module/cam/batch/service/impl/CuBatchExtractServiceImpl.java:
```diff
@@ -337,7 +346,53 @@ public HashSet<PurchasingAccountsPayableDocument> savePOLines(final List<Entry>
-    
+
+    /*
+     * Copied this private KualiCo method from the 2023-10-18 financials version of the superclass.
```
If I remember correctly, I added this for the FINP-10196 backport. I'll update the comments to include the FINP number.

db483 replied:
Thank you

cah292 commented on src/main/java/edu/cornell/kfs/module/cam/batch/service/impl/CuBatchExtractServiceImpl.java:
```diff
@@ -280,9 +282,16 @@ public HashSet<PurchasingAccountsPayableDocument> savePOLines(final List<Entry>
-                            assetAccount = createPurchasingAccountsPayableLineAssetAccount(currentEntry, cabPurapDoc, purApAccountingLine, itemAsset);
-                            assetAcctLines.put(acctLineKey, assetAccount);
-                            itemAsset.getPurchasingAccountsPayableLineAssetAccounts().add(assetAccount);
+                            assetAccount = createPurchasingAccountsPayableLineAssetAccount(currentEntry, cabPurapDoc,
```
Yes, those changes are related to FINP-10196. I'll add comments to mark that accordingly.

I believe the FINP-10196 changes are a follow-up to the FINP-9914 changes from the 2023-08-30 patch. If you think they should be committed separately, then they can't be added until the 2023-08-30 upgrade has been merged to develop. Do you still want such a separation, or do you think it makes sense to keep the changes together?

db483 replied:
Thank you for explaining this. It was not marked by KualiCo as a dependency in the release notes. In this case, I am ok with the fix being added with the upgrade.

db483 commented on src/main/java/edu/cornell/kfs/rass/batch/RassBooleanConverter.java:
```diff
@@ -0,0 +1,13 @@
+
+import org.kuali.kfs.krad.bo.PersistableBusinessObject;
+
+public class RassBooleanConverter extends RassValueConverterBase {
```
Thank you for explaining this.

cah292 commented:
I believe I've addressed all the areas needing attention. I also updated a few XML files to provide bean constructor args explicitly, and had to modify one of the recent Concur fixes for compatibility with a newer OpenCSV version.

This should be ready for review again.

db483 commented:
Thank you for explaining the additional changes.
